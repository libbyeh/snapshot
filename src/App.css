

.App {
  text-align: center;
}

.App {
  background-image: url('./images/white.jpg');
  background-size: cover;
  bottom: 0;
  left: 0;
  overflow: auto;
  position: absolute;
  right: 0;
  top: 0;
}

/*Header*/
.header {
  background-color: none;
  border-bottom: 5px double #555;
  height: 190px;
  overflow: hidden;
  width: 90%;
  z-index: 3;
  margin-left: 65px;
}

.header-title {
  margin-top: 18px;
  margin-bottom: 35px;
  font-family: 'Open Sans', sans-serif;
  color: #555;
  text-decoration: none;
  font-size: 100px;
  font-weight: 800;
  letter-spacing: -3px;
  line-height: 1;
  text-shadow: #EDEDED 3px 2px 0;
  position: relative;
}

.subhead {
  height: 50px;
  color: #555;
  font-family: 'Roboto Condensed', sans-serif;
}

/*Intro-Section*/
.snapshot-intro {
  margin-top: 0px;
  display: flex;
}

.polaroid-image-main {
  background-image: url('./images/polaroid.jpg');
  background-size: cover;
  position: relative;
  z-index: 2;
  height: 508px;
  width: 415px;
  margin: 35px 80px 0px;
  border-top: 10px solid white;
  border-color: #d5d5d5;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.box {
  width: 95.8%;
  background: -moz-linear-gradient(top, rgba(69,72,77,1) 0%, rgba(0,0,0,1) 56%, rgba(0,0,0,1) 95%); 
  background: -webkit-linear-gradient(top, rgba(69,72,77,1) 0%,rgba(0,0,0,1) 56%,rgba(0,0,0,1) 95%); 
  color: #FFF;
  position: relative;
  margin: -30px 8.5px 0px;
  z-index: 7;
  height: 390px;
}

.intro-details {
  margin-top: 50px;
  margin-left: 30px;
  width: 400px;
  text-align: left;
}

.intro-headline {
  color: #A5832B;
  padding-top: 50px;
  width: 275px;
  font-size: 1.8em;
  font-family: 'Open Sans', sans-serif;
}


/*Flashcard Set Options*/
.flashcard-selections {
  color: #A5832B;
  height: 50px;
  margin-top: 110px;
  font-family: 'Open Sans', sans-serif;
  font-weight: bold;
  font-size: 1.8em;
  margin-left: 180px;
  text-align: right;
}

.flashcard-container {  
  display: flex;
  margin-left: 100px;
} 

.individual-flashcard-option {
  display: block;
} 

.flashcard-title-box {
  height: 140px;
  width: 230px;
  background-color: #387b89;
  color: white;
  text-align: right;
  font-size: 1.1em;
  padding-top: 20px;
  padding-right: 15px;
  margin-left: 0px;
  margin-top: 30px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.start-flashcards {
  margin-bottom: 190px;
}

.flashcard-image {
  background-image: url('./images/proto4.png');
  background-size: cover;
  margin-top: 30px;
  height: 149.5px;
  width: 150px;
  position: relative;
  z-index: 2;
  border: 5px solid white;
  background-color: white;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.flashcard-image-2 {
  background-image: url('./images/star.png');
  background-size: cover;
  margin-top: 30px;
  height: 149.5px;
  width: 150px;
  position: relative;
  z-index: 2;
  border: 5px solid white;
  background-color: white;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}



