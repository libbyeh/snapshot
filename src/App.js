import React, { Component } from 'react';
import './App.css';
import Flashcards from './Flashcards';


class App extends Component {
  constructor() {
    super();
    this.state = {
      snapshot: []
    }
  }

  componentDidMount(){
  fetch('http://memoize-datasets.herokuapp.com/api/v1/snapshot')
    .then(response => response.json())
    .then(data => {
      this.setState({
        snapshot: data.snapshot,
      })
    })
    .catch(error => console.log('first fetch', error))
  }

  render() {
    return (
      <div className="App">
        <header className='header'>
          <h1 className='header-title'>[snap-shot]</h1>
        </header>
        <section className='flashcard-intro'>
          <div className='intro'>
            <h2 className='intro-headline'>Flashcards for advanced learning of programming languages and concepts.</h2>
            <p className='scroll-down'>Get Started!</p>
            <p className='arrow'>&#x25BC;</p>
          </div>
          <div className='intro-hero'>
            <div className='polaroid-image'>
              <div className="box">
                <div className="main-headline">[snap-shot]</div>
                <section className='subhead'>
                  <p className='sub1'>noun</p>
                  <p className='sub2'>1. a brief appraisal, summary, or profile</p>
                  <p className='sub3'>2. a Software Developer Learning Tool</p>
                </section>
              </div>
            </div>
          </div>
        </section>
       {
        this.state.snapshot.map((arr) => {
          return (
            <div className='flashcard-body'>
              <div className='polaroid-image'>
                <div className='array-cards'>
                  <h2 className='flashcard-definition'>{arr.definition}</h2>
                  <input></input>
                  <button className='answer-button'>Check Answer</button>
                </div>
              </div>
            </div>
          )
        })
       }
      </div>
    );
  }
}

export default App;
