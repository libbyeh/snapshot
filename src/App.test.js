import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

const snapshot = [
  {
    'id': 1,
    'answer': "Array.length",
    'definition': "Sets or returns the number of elements in that array.",
    'answerCorrect': "false",
    'link': "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length",
  }
]


describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <App key= 'App'/>
      )
    wrapper.setState({
      snapshot: snapshot
    })
  })
  

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update the state when renderFlashcards is called', () => {
    wrapper.instance().renderFlashcards();
    expect(wrapper.instance().state.flashCards).toEqual(snapshot);
    expect(wrapper.instance().state.canRemoveFlashcard).toEqual(false);
  })


  it('should update the state when resetFlashcards is called', () => {
    const mockEvent = {
     preventDefault() {},
     target: { classList: { contains: jest.fn() } }
   }
  wrapper.instance().resetFlashcards(mockEvent)
  expect(wrapper.instance().state.flashCards).toEqual('')
  expect(wrapper.instance().state.canRemoveFlashcard).toEqual(false);
})


})
