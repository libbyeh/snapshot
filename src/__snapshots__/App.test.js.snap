// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <h1
      className="header-title"
    >
      <span>
        [snap-
      </span>
      shot
      <span>
        ]
      </span>
    </h1>
  </header>
  <section
    className="snapshot-intro"
  >
    <div
      className="flashcard-selections"
    >
      Browse Languages & Concepts:
      <div
        className="individual-flashcard-option"
      >
        <div
          className="flashcard-container"
        >
          <div
            className="flashcard-image"
          />
          <div
            className="flashcard-title-box"
          >
            Array Prototypes
            <button
              className="start-flashcards"
              onClick={[Function]}
            >
              Start Learning
            </button>
          </div>
        </div>
        <div
          className="flashcard-container"
        >
          <div
            className="flashcard-image-2"
          />
          <div
            className="flashcard-title-box"
          >
            My Saved Concepts
            <button
              className="start-flashcards"
              onClick={[Function]}
            >
              Start Practicing
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="intro-hero"
    >
      <div
        className="polaroid-image-main"
      >
        <div
          className="box"
        >
          <div
            className="intro-details"
          >
            <h2
              className="intro-headline"
            >
              Flashcards for advanced learning of programming languages and concepts.
            </h2>
          </div>
        </div>
      </div>
    </div>
  </section>
  <Flashcards
    canRemoveFlashcard={false}
    chosenFlashcardSet=""
    removeFlashcardFromSet={Array []}
    renderSavedFlashcards={[Function]}
    resetFlashcards={[Function]}
  />
  <Answer
    chosenFlashcardSet=""
  />
</div>
`;
